src/Analyzers/CSharp/Tests/UseAutoProperty/UseAutoPropertyTests.cs

```diff
@@ -2447,6 +2447,40 @@ int P
             return i;
         }
     }
+}");
+        }
+
+        [WorkItem(47999, "https://github.com/dotnet/roslyn/issues/47999")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseAutoProperty)]
+        public async Task TestPropertyIsReadOnlyAndSetterNeeded()
+        {
+            await TestInRegularAndScript1Async(
+@"class Class
+{
+    [|int i|];
+    public readonly int P => i;
+    public void SetP(int value) => i = value;
+}",
+@"class Class
+{
+    public int P { get; private set; }
+    public void SetP(int value) => P = value;
+}");
+        }
```

Comment:
can you add a test without hte accessibility modifier?  i.e. `readonly int P => 1;`  I want to make sure removing that modifier doesn't lead to very weirdly formatted code.  thanks!